src/zenml/stack_stores/base_stack_store.py

```diff
@@ -304,7 +328,55 @@ def __check_component(
             typ: name for typ, name in map(__check_component, stack.components)
         }
         metadata = {c.type.value: c.flavor for c in stack.components}
-        self._create_stack(stack.name, stack_configuration)
+        self._save_stack(stack.name, stack_configuration)
+        logger.info("Registered stack with name '%s'.", stack.name)
+        return metadata
+
+    def update_stack(self, name: str, stack: StackWrapper) -> Dict[str, str]:
+        """Update a stack and its components.
+
+        If any of the stack's components aren't registered in the stack store
+        yet, this method will try to register them as well.
+
+        Args:
+            name: The original name of the stack.
+            stack: The new stack to use in the update.
+
+        Returns:
+            metadata dict for telemetry or logging.
+
+        Raises:
+            DoesNotExistException: If no stack exists with the given name.
+        """
+        try:
+            self.get_stack(name)
+        except KeyError:
+            raise DoesNotExistException(
```

Comment:
I would keep this as a `KeyError` (and get rid of the DoesNotExistException entirely) as we use that in all other cases as well.